$NetBSD: patch-bb,v 1.7 2011/07/23 01:14:44 manu Exp $

O_DIRECTORY is defined as 0, therefore ((flags & O_DIRECTORY) == O_DIRECTORY)
is always true even if the flag is not set.

--- xlators/performance/quick-read/src/quick-read.c.orig	2011-07-14 14:21:10.000000000 +0200
+++ xlators/performance/quick-read/src/quick-read.c	2011-07-22 09:16:44.000000000 +0200
@@ -727,9 +727,9 @@
                 }
         }
         UNLOCK (&table->lock);
 
-        if (content_cached && ((flags & O_DIRECTORY) == O_DIRECTORY)) {
+        if (content_cached && (flags & O_DIRECTORY)) {
                 op_ret = -1;
                 op_errno = ENOTDIR;
                 gf_log (this->name, GF_LOG_WARNING,
                         "open with O_DIRECTORY flag received on non-directory");
